tests/conftest.py

```diff
@@ -144,3 +146,36 @@ def mock_ic13(tmpdir_factory, mock_image_stream):
         with open(fn_i, 'wb') as f:
             f.write(file.getbuffer())
     return str(image_folder), str(label_folder)
+
+
+@pytest.fixture(scope="session")
+def mock_svhn_dataset(tmpdir_factory, mock_image_stream):
+    root = tmpdir_factory.mktemp('datasets')
+    svhn_root = root.mkdir('svhn')
+    file = BytesIO(mock_image_stream)
+    # ascii image names
+    first = np.array([[49], [46], [112], [110], [103]], dtype=np.int16)  # 0.png
+    second = np.array([[50], [46], [112], [110], [103]], dtype=np.int16)  # 1.png
+    third = np.array([[51], [46], [112], [110], [103]], dtype=np.int16)  # 2.png
+    # labels: label is also ascii
+    label = {'height': [35, 35, 35, 35], 'label': [1, 1, 3, 7],
+             'left': [116, 128, 137, 151], 'top': [27, 29, 29, 26],
+             'width': [15, 10, 17, 17]}
+
+    matcontent = {'digitStruct': {'name': [first, second, third], 'bbox': [label, label, label]}}
+    # mock train data
+    root = svhn_root.mkdir('train')
+    hdf5storage.write(matcontent, filename=root.join('digitStruct.mat'))
+    for i in range(4):
+        fn = root.join(f'{i}.png')
+        with open(fn, 'wb') as f:
+            f.write(file.getbuffer())
+    # mock test data
+    root = svhn_root.mkdir('test')
+    hdf5storage.write(matcontent, filename=root.join('digitStruct.mat'))
+    for i in range(4):
+        fn = root.join(f'{i}.png')
+        with open(fn, 'wb') as f:
+            f.write(file.getbuffer())
```

Comment:
I didn't realize but for unittests, we don't need to have a train & test version: since we monkeypatch the constructor, it won't change anything (you can remove the `train=False` test case as well)